### PR TITLE
Feature: OpenCTI Improvements

### PR DESCRIPTION
**Pull request by jesusgpo**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

In progess doing test, waiting for paloalto review

## Description
- Rename indicators commands to observable commands according to [OpenCTI naming](https://docs.opencti.io/latest/usage/exploring-observations/) the deprecation
  - opencti-get-indicators -> opencti-get-observables
  - opencti-indicator-delete -> opencti-observable-delete
  - opencti-indicator-field-update -> opencti-observable-field-update
  - opencti-indicator-create -> opencti-observable-create
  - opencti-indicator-field-add -> opencti-observable-field-add
  - opencti-indicator-field-remove -> opencti-observable-field-remove
- Develop Incident commands
  - opencti-incident-create
  - opencti-incident-delete
  - opencti-incident-types-list
- Develop indicators commands
  - opencti-indicator-create
  - opencti-indicator-update
  - opencti-indicator-field-add
  - opencti-indicator-field-remove
  - opencti-get-indicators
  - opencti-indicator-types-list
  - opencti-relationship-create: Required to create entities relations

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by samuelFain** [comment]
@jesusgpo Following our sync on this contribution:
1. Please make sure you sign the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form so I can start my review on this PR.
2. We are currently reviewing the original commands depreciation issue and whether we should keep or completely remove them, I will update you as soon as I have more information.

Thanks!

**Comment by samuelFain** [comment]
@jesusgpo Following our sync on this contribution:
> Hi Samuel, I have just completed the form and I filled the description with the commands developed and updated, my next step is develop the tests for the commands

Thanks for the update! I will start reviewing the PR.

File: Packs/OpenCTI/Integrations/OpenCTI/OpenCTI.yml [hunk 1]
```diff
@@ -293,12 +300,335 @@ script:
...
+      description: New Relationship Entity Type.
+  - arguments: []
+    description: Get a list of all indicator types.
+    name: opencti-indicator-types-list
+    outputs:
+    - contextPath: OpenCTI.IndicatorTypes.IndicatorTypesList.id
+      description: Indicator type ID.
+    - contextPath: OpenCTI.IndicatorTypes.IndicatorTypesList.name
+      description: Indicator type name.
+    - contextPath: OpenCTI.IndicatorTypes.ndicatorTypesList.description
```
Comment (samuelFain) [inline Packs/OpenCTI/Integrations/OpenCTI/OpenCTI.yml:624]:
```suggestion
    - contextPath: OpenCTI.IndicatorTypes.IndicatorTypesList.description
```

**Review by samuelFain — CHANGES_REQUESTED** [review]
@jesusgpo You have done amazing work so far!
I reviewed most of the command changes so I'm posting my initial review for them.
I will be reviewing the incident commands (3 commands), test file and other file changes in this PR shortly.
In the meanwhile, please see my comments, some of them might have already been addressed by your latests commits.
Also, it seems like you have a conflict in README.md.

**Comment by samuelFain** [comment]
@jesusgpo A few more issues that came up when running demisto-sdk pre-commit command on your PR (I strongly suggest to use it as well, more about this command [here](https://docs-cortex.paloaltonetworks.com/r/1/Demisto-SDK-Guide/pre-commit)):
- ruff error code [E731](https://docs.astral.sh/ruff/rules/lambda-assignment/): Do not assign a `lambda` expression, use a `def`. For example in: `query = lambda *args, **kwargs: None`.
- Getting a mypy linting error where the `indicator` argument has a type of `Optional[str]` but expects `str` for this line:
`pattern = build_stix_pattern(indicator, main_observable_type)` 
- I also have 3 UTs failing when trying to run them via pytest on the docker container, not sure if there is real issue with them or not, we can wait until the conflict in `Packs/OpenCTI/Integrations/OpenCTI/README.md` is resolved and then pre-commit will also run inside one of this PRs Github actions (also runs the UTs), giving us something to compare with:
test_incident_create_command_exception, test_incident_types_list_command_exception, 
OpenCTI_test.py::test_incident_create_command_exception_id.

**Comment by samuelFain** [comment]
@jesusgpo the latest changes are looking good! i've sent you a slack massage regarding the current error messages. mostly small discrepancy in commands between the .yml and .py files, and updating the test playbook to match the changes in this PR.
Please let me know if any issues arise.

File: Packs/OpenCTI/Integrations/OpenCTI/OpenCTI.yml [hunk 2]
```diff
@@ -293,12 +300,350 @@ script:
...
+      name: create_observables
+      predefined:
+      - "true"
+      - "false"
+    description: Create a new indicator in OpenCTI.
+    name: opencti-indicator-create
+    outputs:
+    - contextPath: OpenCTI.Indicator.id
+      description: New Indicator ID.
+      type: string
```
Comment (samuelFain) [inline Packs/OpenCTI/Integrations/OpenCTI/OpenCTI.yml:430]: Regarding the following validation error RM102:
>opencti-indicator-create:
The following outputs are missing from yml: OpenCTI.Indicator.value, OpenCTI.Indicator.type

The validation's description validates that all context outputs defined in the README file are present in the YML file, and vice versa.
Where the rational is:
>"Ensuring consistency between the README and YML files helps maintain accurate documentation and prevents discrepancies."

So basically there is a consistency between the readme:
https://github.com/demisto/content/blob/bf19c98a05b0af567658e6e6624091b3cda295d6/Packs/OpenCTI/Integrations/OpenCTI/README.md?plain=1#L154
And the yml file which only have OpenCTI.Indicator.id in the context output:
https://github.com/demisto/content/pull/37270/files#diff-3a03284b240156d148b67fe01f470c1da9010d62638df25c8b68e4b5e184b4b2R428

It also relevant for `opencti-get-indicators`:
>opencti-get-indicators:
The following outputs are missing from readme: OpenCTI.Indicators.IndicatorList.creators

**Comment by samuelFain** [comment]
@jesusgpo Great work so far!
As discussed on Slack, we will schedule a demo call to review the changes made in this PR before merging this contribution.

**Comment by Shellyber** [comment]
For the Reviewer: Trigger build request has been accepted for this contribution PR.

**Comment by Shellyber** [comment]
For the Reviewer: Successfully created a pipeline in GitLab with url: https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/1788990

File: Packs/OpenCTI/Integrations/OpenCTI/README.md [hunk 3]
```diff
@@ -630,3 +523,350 @@ Get a list of all marking definitions.
...
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| name | Incident name. | Required | 
+| incident_type | Incident Type name. Use opencti-incident-types-list to find all incident types in OpenCTI. | Optional | 
+| confidence | Incident Confidence Number. Values range is 0 - 100. Default value is 50. | Optional | 
+| severity | Incident severity. Possible values are: low, medium, high, critical. | Optional | 
+| description | Incident description. | Optional | 
+| source | Incident Source. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/OpenCTI/Integrations/OpenCTI/README.md:641]:
```suggestion
| source | Incident source. | Optional | 
```

File: Packs/OpenCTI/Integrations/OpenCTI/README.md [hunk 4]
```diff
@@ -630,3 +523,350 @@ Get a list of all marking definitions.
...
+| name | Incident name. | Required | 
+| incident_type | Incident Type name. Use opencti-incident-types-list to find all incident types in OpenCTI. | Optional | 
+| confidence | Incident Confidence Number. Values range is 0 - 100. Default value is 50. | Optional | 
+| severity | Incident severity. Possible values are: low, medium, high, critical. | Optional | 
+| description | Incident description. | Optional | 
+| source | Incident Source. | Optional | 
+| objective | Incident objective. | Optional | 
+| created_by | Organization ID. Use opencti-organization-list to find all organization IDs in OpenCTI, or use opencti-organization-create to create a new organization ID. | Optional | 
+| first_seen | Incident First seen. YYYY-MM-DDThh:mm:ss.sssZ. | Optional | 
+| last_seen | Incident Last seen. YYYY-MM-DDThh:mm:ss.sssZ. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/OpenCTI/Integrations/OpenCTI/README.md:645]:
```suggestion
| first_seen | Incident first seen. YYYY-MM-DDThh:mm:ss.sssZ. | Optional | 
| last_seen | Incident last seen. YYYY-MM-DDThh:mm:ss.sssZ. | Optional | 
```

File: Packs/OpenCTI/Integrations/OpenCTI/README.md [hunk 5]
```diff
@@ -630,3 +523,350 @@ Get a list of all marking definitions.
...
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| search | The incident's value to filter by, can be partial value. | Optional | 
+| created_by | The ID of the entity that created the incident (use opencti-organization-list to find or create). | Optional | 
+| creator | The ID of the incident creator. | Optional | 
+| created_after | Created after date filter. Format: YYYY-MM-DDThh:mm:ss.sssZ. | Optional | 
+| created_before | Created before date filter. Format: YYYY-MM-DDThh:mm:ss.sssZ. | Optional | 
+| incident_types | The types of the incident. Use opencti-incident-types-list to find all incident types in OpenCTI. | Optional | 
+| label_id | The label ID for the incident (use opencti-label-list to find or create). | Optional | 
+| limit | The maximum number of incidents to return. Default value is 50. Maximum value is 500. Default is 50. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/OpenCTI/Integrations/OpenCTI/README.md:676]:
```suggestion
| limit | The maximum number of incidents to return. Maximum value is 500. Default is 50. | Optional | 
```

File: Packs/OpenCTI/Integrations/OpenCTI/README.md [hunk 6]
```diff
@@ -630,3 +523,350 @@ Get a list of all marking definitions.
...
+| search | The incident's value to filter by, can be partial value. | Optional | 
+| created_by | The ID of the entity that created the incident (use opencti-organization-list to find or create). | Optional | 
+| creator | The ID of the incident creator. | Optional | 
+| created_after | Created after date filter. Format: YYYY-MM-DDThh:mm:ss.sssZ. | Optional | 
+| created_before | Created before date filter. Format: YYYY-MM-DDThh:mm:ss.sssZ. | Optional | 
+| incident_types | The types of the incident. Use opencti-incident-types-list to find all incident types in OpenCTI. | Optional | 
+| label_id | The label ID for the incident (use opencti-label-list to find or create). | Optional | 
+| limit | The maximum number of incidents to return. Default value is 50. Maximum value is 500. Default is 50. | Optional | 
+| last_run_id | The last ID from the previous call, from which to begin pagination for this call. You can find this value at the OpenCTI.Incidents.LastRunID context path. | Optional | 
+| all_results | When the argument is set to true, the limit argument is ignored. Default is false. Possible values are: true, false. Default is false. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/OpenCTI/Integrations/OpenCTI/README.md:678]:
```suggestion
| all_results | When the argument is set to true, the limit argument is ignored. Possible values are: true, false. Default is false. | Optional | 
```

File: Packs/OpenCTI/ReleaseNotes/2_0_0.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+
+#### Integrations
+
+##### OpenCTI
+
+- Deprecate old `indicator` commands
```
[the remark this answers is not in the document]
Comment (samuelFain) [reply Packs/OpenCTI/ReleaseNotes/2_0_0.md:6]: @ShirleyDenkberg is this a mistake?

**Review by samuelFain — APPROVED** [review]
Amazing work! Approved.

**Pull request merged**